lib/src/browser_adapter.dart

```diff
@@ -28,6 +29,21 @@ import 'package:w_transport/src/web_socket/w_socket.dart';
 /// classes that return browser-specific implementations that leverage
 /// dart:html.
 class BrowserAdapter implements PlatformAdapter {
+  final bool _useSockJS;
+  final bool _sockJSDebug;
+  final bool _sockJSNoCredentials;
+  final List<String> _sockJSProtocolsWhitelist;
+
+  BrowserAdapter(
+      {bool useSockJS,
+      bool sockJSNoCredentials,
+      bool sockJSDebug: false,
+      List<String> sockJSProtocolsWhitelist})
+      : _useSockJS = useSockJS == true,
+        _sockJSDebug = sockJSDebug == true,
```

Comment:
This is one thing I've discovered with this layered approach (abstract interface --> platform adapter --> implementation).. if each one takes the same optional params, the proxying methods at the top just pass on whatever they received. This means that if the end consumer doesn't supply a value for an optional param, it will get passed through to the next layer as `null`, which overrides the default value.